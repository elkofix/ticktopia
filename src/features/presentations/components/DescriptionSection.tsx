export function DescriptionSection({ description }: { description: string }) {
  return (
    <div 
      className="bg-white p-6 rounded-xl shadow-md mb-12"
      data-testid="description-section"
    >
      <h2 className="text-2xl font-bold text-gray-900 mb-4">Descripci√≥n</h2>
      <p 
        className="text-gray-700 whitespace-pre-line"
        data-testid="description-content"
      >
        {description}
      </p>
    </div>
  );
}