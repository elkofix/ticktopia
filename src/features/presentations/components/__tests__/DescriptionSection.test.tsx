import React from 'react';
import { render, screen } from '@testing-library/react';
import { DescriptionSection } from '../DescriptionSection';

describe('DescriptionSection', () => {
  const testDescription = "This is a test description.\nWith multiple lines.";

  it('renders correctly with description', () => {
    render(<DescriptionSection description={testDescription} />);
    
    // Check for the title
    expect(screen.getByText('Descripción')).toBeInTheDocument();
    expect(screen.getByText('Descripción')).toHaveClass('text-2xl', 'font-bold', 'text-gray-900', 'mb-4');
    
    // Check for the description content
  });

  it('handles empty description', () => {
    render(<DescriptionSection description="" />);
    
    expect(screen.getByText('Descripción')).toBeInTheDocument();
    expect(screen.getByTestId('description-content').textContent).toBe('');
  });

  it('preserves line breaks with whitespace-pre-line', () => {
    render(<DescriptionSection description={testDescription} />);
    
    const content = screen.getByTestId('description-content');
    expect(content).toHaveClass('whitespace-pre-line');
  });

  it('matches snapshot with description', () => {
    const { asFragment } = render(<DescriptionSection description={testDescription} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot with empty description', () => {
    const { asFragment } = render(<DescriptionSection description="" />);
    expect(asFragment()).toMatchSnapshot();
  });
});