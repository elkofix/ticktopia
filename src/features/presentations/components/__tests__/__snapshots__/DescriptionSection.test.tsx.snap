// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DescriptionSection matches snapshot with description 1`] = `
<DocumentFragment>
  <div
    class="bg-white p-6 rounded-xl shadow-md mb-12"
    data-testid="description-section"
  >
    <h2
      class="text-2xl font-bold text-gray-900 mb-4"
    >
      Descripción
    </h2>
    <p
      class="text-gray-700 whitespace-pre-line"
      data-testid="description-content"
    >
      This is a test description.
With multiple lines.
    </p>
  </div>
</DocumentFragment>
`;

exports[`DescriptionSection matches snapshot with empty description 1`] = `
<DocumentFragment>
  <div
    class="bg-white p-6 rounded-xl shadow-md mb-12"
    data-testid="description-section"
  >
    <h2
      class="text-2xl font-bold text-gray-900 mb-4"
    >
      Descripción
    </h2>
    <p
      class="text-gray-700 whitespace-pre-line"
      data-testid="description-content"
    />
  </div>
</DocumentFragment>
`;
